lib/themes/dosomething/paraneue_dosomething/js/components/Slideshow.js

```diff
@@ -11,27 +11,53 @@ class Slideshow extends React.Component {
     super(props);
 
     this.state = {
-      activeSlide: 0
+      currentSlide: 0,
+      animationDirection: 'left'
     };
+
+    this.moveBackward = this.moveBackward.bind(this);
+    this.moveForward = this.moveForward.bind(this);
   }
 
-  moveSlide() {
-    //
+  moveBackward() {
+    const slide = this.state.currentSlide - 1;
+
+    this.setState({
+      currentSlide: slide,
+      animationDirection: 'right'
+    });
+  }
+
+  moveForward() {
+    const slide = this.state.currentSlide + 1;
+
+    this.setState({
+      currentSlide: slide,
+      animationDirection: 'left'
+    });
   }
 
   render() {
     const user = this.props.user;
     const campaign = this.props.campaign;
 
     const slides = this.props.slides.map((slide, index) => {
-      return React.createElement(slide, {key: index, campaign: campaign, user: user});
+      const isActive = this.state.currentSlide == index;
+      const displayClasses = ['slideshow__slide'];
```

Comment:
This seems iffy to me. The slides shouldn't require a `Slideshow` in order to be styled as slides. Ideally the slideshow just passes props (either booleans or enum-style values via a PropTypes contract) rather than explicitly applying classes to its children.